### DISCUSSION TITLE
Conditional keymaps

### DISCUSSION THREAD

**Discussion by kuntau** [wezterm/wezterm#1565] [Q&A] [answered]
First of all thank you for this amazing software. I've been using wezterm exclusively for a few weeks now.

I always have neovim running in wezterm and I'm looking for more efficient way to navigate neovim pane & wezterm pane.

Coming from **tmux**, I used to use [vim-tmux-navigator](https://github.com/christoomey/vim-tmux-navigator) to make my life easier.
Basically, it's <kbd>ctrl</kbd>+<kbd>h</kbd><kbd>j</kbd><kbd>k</kbd><kbd>l</kbd> to navigate all pane and the plugin will check whether I'm in vim or not.

Is it possible to do something like this in wezterm?

I have this following keymaps in wezterm

```lua
{ key="l", mods="CTRL", action=wezterm.action{ActivatePaneDirection="Right"} },
{ key="h", mods="CTRL", action=wezterm.action{ActivatePaneDirection="Left"} },
{ key="k", mods="CTRL", action=wezterm.action{ActivatePaneDirection="Up"} },
{ key="j", mods="CTRL", action=wezterm.action{ActivatePaneDirection="Down"} },
```

The simplest I'am trying to do is to detect whether the pane I'm in is zoomed in or not. And don't set keymaps above if I'm in zoomed pane (meaning I'm in vim). As for now if I'm in zoomed vim, I can't navigate vim pane, even if I have set `  unzoom_on_switch_pane = false`.

Thank you.

**Comment by wez** [comment, marked as answer]
https://github.com/wez/wezterm/discussions/1458 is a larger example of this sort of thing.
The gist of it is something like this:

```lua
-- Equivalent to POSIX basename(3)
-- Given "/foo/bar" returns "bar"
-- Given "c:\\foo\\bar" returns "bar"
function basename(s)
  return string.gsub(s, "(.*[/\\])(.*)", "%2")
end

function is_vim(pane)
   local exe = basename(pane:get_foreground_process_name())
   return exe == "vim"
end

return {
  keys = {
    {key="l", mods="CTRL",
      action=wezterm.action_callback(function(win, pane)
           if is_vim(pane) then
              -- send something to vim. Not sure if you just wanted
              -- to pass the ctrl-L through, but that's what this does
              win:perform_action{SendKey={key="l", mods="CTRL"}}
           else
              win:perform_action{ActivatePaneDirection="Right"}}
           end
      end)
    }
}
```

Check out:
https://wezfurlong.org/wezterm/config/lua/pane/get_foreground_process_name.html
https://wezfurlong.org/wezterm/config/lua/window/perform_action.html
https://wezfurlong.org/wezterm/config/lua/wezterm/action_callback.html
